integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java

```diff
@@ -0,0 +1,131 @@
+
+public class QueryAnalysisIT extends PPLIntegTestCase {
+
+    @Override
+    public void init() throws IOException {
+        loadIndex(Index.ACCOUNT);
+    }
+
+    /**
+     * Valid commands should pass semantic check
+     */
+    @Test
+    public void searchCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s age=20", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Test
+    public void whereCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | where age=20", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Test
+    public void fieldsCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | fields firstname", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Ignore("Can't resolve target field yet")
+    @Test
+    public void renameCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | rename firstname as first", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Test
+    public void statsCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | stats avg(age)", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Test
+    public void dedupCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | dedup firstname, lastname", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Test
+    public void sortCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | sort age", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Test
+    public void evalCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | eval age=abs(age)", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+
+    /**
+     * Commands that fail semantic analysis should throw {@link SemanticCheckException}
+     */
+    @Test
+    public void unsupportedAggregationShouldFailSemanticCheck() {
+        String query = String.format("search source=%s | stats count(age)", TEST_INDEX_ACCOUNT);
+        queryShouldThrowSemanticException(query, "Unsupported aggregation function count");
+    }
+
+    @Test
+    public void nonexistentIndexShouldFailSemanticCheck() {
+        String query = String.format("search source=%s | fields name", TEST_INDEX_ACCOUNT);
+        queryShouldThrowSemanticException(query, "can't resolve expression name in type env");
+    }
+
+
+    private void queryShouldPassSemanticCheck(String query) {
+        Request request = buildRequest(query);
+        try {
+            Response response = client().performRequest(request);
+            assertEquals(OK.getStatus(), response.getStatusLine().getStatusCode());
```

Comment:
This is same as `executeQuery`? Probably we can just call it and discard return value?

```
    protected String executeQuery(String query) throws IOException {
        Response response = client().performRequest(buildRequest(query));
        Assert.assertEquals(200, response.getStatusLine().getStatusCode());
        return getResponseBody(response, true);
    }
```

Reply:
Done, thanks!